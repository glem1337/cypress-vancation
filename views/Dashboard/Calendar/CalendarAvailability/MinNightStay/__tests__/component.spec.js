import React from 'react';
import { shallow } from 'enzyme';

import MinNightStayComponent from '../component';

describe('MinNightStayComponent component tests', () => {
  const props = {
    toggleOpenedState: jest.fn(),
    isOpened: true,
    handleSubmit: jest.fn(),
    isLoading: false,
    submitForm: jest.fn(),
    isValid: true,
    isBackPopoverVisible: true,
    isDiscardPopoverVisible: true,
    showPopoverOrBack: jest.fn(() => jest.fn()),
    closePopover: jest.fn(() => jest.fn()),
    closePopoversAndBack: jest.fn(() => jest.fn()),
  };

  const component = shallow(<MinNightStayComponent {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
