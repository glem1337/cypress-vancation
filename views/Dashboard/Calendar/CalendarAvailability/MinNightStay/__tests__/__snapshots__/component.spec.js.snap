// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinNightStayComponent component tests matches snapshot 1`] = `
<div
  className="calendar-listing__side-main-item"
>
  <Button
    className="ant-btn-slideout"
    onClick={[MockFunction]}
  >
    <i
      className="icon icon-moon"
    />
    <span>
      <Memo(FormattedMessage)
        id="shared.minimumNightStay"
      />
    </span>
    <i
      className="icon icon-right ml-auto"
    />
  </Button>
  <div
    className="calendar-listing__side-slideout calendar-listing__side-slideout--active"
  >
    <ForwardRef(InternalForm)
      layout="vertical"
    >
      <div
        className="calendar-listing__side-slideout__main"
      >
        <div
          className="mb-16"
        >
          <BackButton
            discardConfirm={[MockFunction]}
            discardPrepare={[MockFunction]}
            discardReject={[MockFunction]}
            isPopoverVisible={true}
          />
        </div>
        <h3
          className="text-subheader mb-16"
        >
          <Memo(FormattedMessage)
            id="calendar.minimumNightStay"
          />
        </h3>
        <Field
          className="main-input__field"
          component={[Function]}
          items={
            Array [
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 1,
                  },
                },
                "value": 1,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 2,
                  },
                },
                "value": 2,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 3,
                  },
                },
                "value": 3,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 4,
                  },
                },
                "value": 4,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 5,
                  },
                },
                "value": 5,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 6,
                  },
                },
                "value": 6,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 7,
                  },
                },
                "value": 7,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 8,
                  },
                },
                "value": 8,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 9,
                  },
                },
                "value": 9,
              },
              Object {
                "label": Object {
                  "id": "addNewCamper.pricing.countOfNights",
                  "values": Object {
                    "count": 10,
                  },
                },
                "value": 10,
              },
            ]
          }
          label={
            Object {
              "id": "addNewCamper.pricing.nights",
            }
          }
          name="minimalNightStay"
          optionLabelProp="label"
          placeholder={
            Object {
              "id": "shared.select",
            }
          }
        />
      </div>
      <FooterComponent
        canSubmit={true}
        discardConfirm={[MockFunction]}
        discardPrepare={[MockFunction]}
        discardReject={[MockFunction]}
        isDiscardPopoverVisible={true}
        isLoading={false}
        isSavePopoverVisible={false}
        submit={[MockFunction]}
        submitBtnText={
          Object {
            "id": "shared.ok",
          }
        }
        withSavePopover={false}
        withSavePopoverWarning={false}
      />
    </ForwardRef(InternalForm)>
  </div>
</div>
`;
